### PR TITLE
codeql/lgtm assorted fixes

### PR DESCRIPTION
**Pull request by jelly**
This is a batch of fixes found by LGTM/codeql, as I don't want to changes to be too invasive / too much to review.

We'll enable codeql by default once all the obvious things are fixed.

### CODE REVIEW

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 1]
```diff
@@ -82,9 +82,9 @@ export class FileAutoComplete extends React.Component {
     }
 
     onCreateOption(newValue) {
-        this.setState({
-            displayFiles: [...this.state.displayFiles, { type: "file", path: newValue }]
-        });
+        this.setState(prevState => ({
+            displayFiles: [...prevState.displayFiles, { type: "file", path: newValue }]
+        }));
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:87]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18008-20221208-125012-6c9d9441-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/firewall.jsx [hunk 2]
```diff
@@ -377,10 +377,10 @@ class AddEditServicesModal extends React.Component {
         }
         p.then(() => Dialogs.close())
                 .catch(error => {
-                    this.setState({
-                        dialogError: this.state.custom ? _("Failed to add port") : _("Failed to add service"),
+                    this.setState(prevState => ({
+                        dialogError: prevState.custom ? _("Failed to add port") : _("Failed to add service"),
```
Comment (cockpituous) [inline pkg/networkmanager/firewall.jsx:381]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18008-20221208-125012-6c9d9441-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/logout-account-dialog.js [hunk 3]
```diff
@@ -44,10 +42,5 @@ export const logoutAccountDialog = (account) => {
         ]
     };
 
-    if (!dlg)
-        dlg = show_modal_dialog(props, footer);
-    else {
-        dlg.setProps(props);
-        dlg.setFooterProps(footer);
-    }
+    show_modal_dialog(props, footer);
```
Comment (cockpituous) [inline pkg/users/logout-account-dialog.js:45]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18008-20221208-125012-6c9d9441-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline pkg/users/logout-account-dialog.js:45]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18008-20221208-172856-e7737fdb-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]

**Review by martinpitt — APPROVED** [review]
Cheers! Let's land this before it bitrots.

**Pull request merged**